src/ClientData/include/ClientData/OwnsCaptureData.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CLIENT_DATA_OWNS_CAPTURE_DATA_H_
+#define CLIENT_DATA_OWNS_CAPTURE_DATA_H_
+
+#include <functional>
+#include <memory>
+#include <optional>
+
+#include "ClientData/CaptureData.h"
+#include "OrbitBase/Logging.h"
+
+namespace orbit_client_data {
+
+class OwnsCaptureData {
+ public:
+  virtual ~OwnsCaptureData() = default;
+
+  // TODO(b/234095077): The two methods should not return ref, as capture_data_ doesn't
+  // necessarily store a value. They should either return a pointer, or
+  // `std::optional<std::reference_wrapper<CaptureData>>`
+  [[nodiscard]] virtual const CaptureData& GetCaptureData() const {
+    ORBIT_CHECK(HasCaptureData());
+    return *capture_data_;
+  }
+  // CallstackDataView needs OrbitApp::GetMutableCaptureData()
+  [[nodiscard]] virtual CaptureData& GetMutableCaptureData() {
+    ORBIT_CHECK(HasCaptureData());
+    return *capture_data_;
+  }
+
+  [[nodiscard]] virtual bool HasCaptureData() const { return static_cast<bool>(capture_data_); }
```

Comment:
IMO, the cast is clearer. I don't like `!=` operator applied to values of different types. How does `capture_data_ != nullptr` read? "Is `capture_data_` not `nullptr`?" How can `capture_data_` ever be `nullptr`? -- it's not a pointer. It can own nothing, though.